src/transforms/operationTransforms.ts

```diff
@@ -478,23 +480,43 @@ function getGroupedParameters(
       !mediaType || !p.targetMediaType || p.targetMediaType === mediaType;
     return Boolean(matchesOperation && matchesMediaType);
   });
-  return {
-    requestBody: operationParams.find(
-      p => p.location === ParameterLocation.Body
-    ),
-    queryParameters: operationParams.filter(
-      p => p.location === ParameterLocation.Query
-    ),
-    urlParameters: operationParams.filter(
-      p =>
-        p.location === ParameterLocation.Path ||
-        p.location === ParameterLocation.Uri
-    ),
-    headerParameters: operationParams.filter(
-      p => p.location === ParameterLocation.Header
-    ),
-    cookie: operationParams.filter(p => p.location === ParameterLocation.Cookie)
-  };
+  if (mediaType && (mediaType == KnownMediaType.Multipart || mediaType == KnownMediaType.Form)) {
+    return {
+      formDataParameters: operationParams.filter(
```

Comment:
you could also do this without the if/else with something like

```typescript
const hasFormDataParameters = mediaType && (mediaType == KnownMediaType.Multipart || mediaType == KnownMediaType.Form);

return {
   ...(hasFormDataParameters  
         ? {formDataParameters: operationParams.filter(p => p.location === ParameterLocation.Body)}
         : {requestBody: operationParams.find(p => p.location === ParameterLocation.Body)} ),
      queryParameters: operationParams.filter(
        p => p.location === ParameterLocation.Query
      ),
      urlParameters: operationParams.filter(
        p =>
          p.location === ParameterLocation.Path ||
          p.location === ParameterLocation.Uri
      ),
      headerParameters: operationParams.filter(
        p => p.location === ParameterLocation.Header
      ),
      cookie: operationParams.filter(p => p.location === ParameterLocation.Cookie)
}
```

Reply:
Done